discord/voice_client.py

```diff
@@ -623,13 +623,33 @@ def get_ssrc(self, user_id):
             user_id
         ]
 
+    @overload
     def play(
         self,
         source: AudioSource,
         *,
         after: Callable[[Exception | None], Any] = None,
-        wait_finish: bool = False,
+        wait_finish: Literal[False] = False,
     ) -> None:
+        ...
+
+    @overload
+    def play(
+        self,
+        source: AudioSource,
+        *,
+        after: Callable[[Exception | None], Any] = None,
+        wait_finish: Literal[True],
+    ) -> asyncio.Future:
+        ...
+
+    def play(
+        self,
+        source: AudioSource,
+        *,
+        after: Callable[[Exception | None], Any] = None,
+        wait_finish: bool = False,
+    ) -> None | asyncio.Future:
```

Comment:
Wouldn't this actually fix the new mypy error?
```suggestion
    ):
```

Reply:
We don't need to fix, it's intended (`None` will be returned and it cannot be awaited)

Reply:
My bad.